lib/ember_cli/shell.rb

```diff
@@ -34,15 +34,17 @@ def stop
 
     def install
       if paths.gemfile.exist?
-        exec "#{paths.bundler} install"
+        run! "#{paths.bundler} install"
       end
 
-      exec "#{paths.npm} prune && #{paths.npm} install"
-      exec "#{paths.bower} prune && #{paths.bower} install"
+      clean_ember_dependencies! if invalid_ember_dependencies?
```

Comment:
> But, but... its so dang readable!

We like to refer to them as `"surprise conditionals"` -- they make lines so "readable" (in an English sense) that they tend to blend in *too* well.

Conditionals (and `if` / `else` blocks) contain logic that depends on program state. They should jump out at developers.